src/anemoi/training/diagnostics/plots.py

```diff
@@ -473,124 +485,93 @@ def plot_flat_sample(
-            err_plot = error_plot_in_degrees(pred, input_)
-            scatter_plot(
-                fig,
-                ax[4],
-                lon=lon,
-                lat=lat,
-                data=err_plot,
-                cmap="bwr",
-                norm=TwoSlopeNorm(vcenter=0.0),
-                title=f"{vname} increment [pred - input] % 360",
-            )
-            err_plot = error_plot_in_degrees(truth, input_)
-            scatter_plot(
-                fig,
-                ax[5],
-                lon=lon,
-                lat=lat,
-                data=err_plot,
-                cmap="bwr",
-                norm=TwoSlopeNorm(vcenter=0.0),
-                title=f"{vname} persist err: {np.nanmean(np.abs(err_plot)):.{4}f} deg.",
-            )
-        else:
-            scatter_plot(fig, ax[0], lon=lon, lat=lat, data=input_, title=f"{vname} input")
-            scatter_plot(
-                fig,
-                ax[4],
-                lon=lon,
-                lat=lat,
-                data=pred - input_,
-                cmap="bwr",
-                norm=TwoSlopeNorm(vcenter=0.0),
-                title=f"{vname} increment [pred - input]",
-            )
-            scatter_plot(
-                fig,
-                ax[5],
-                lon=lon,
-                lat=lat,
-                data=truth - input_,
-                cmap="bwr",
-                norm=TwoSlopeNorm(vcenter=0.0),
-                title=f"{vname} persist err",
-            )
+        single_plot(fig, ax[0], lon, lat, input_, title=f"{vname} input", scatter=scatter)
+        single_plot(
+            fig,
+            ax[4],
+            lon,
+            lat,
+            pred - input_,
+            cmap="bwr",
+            norm=TwoSlopeNorm(vcenter=0.0),
+            title=f"{vname} increment [pred - input]",
+            scatter=scatter,
+        )
+        single_plot(
+            fig,
+            ax[5],
+            lon,
+            lat,
+            truth - input_,
+            cmap="bwr",
+            norm=TwoSlopeNorm(vcenter=0.0),
+            title=f"{vname} persist err",
+            scatter=scatter,
+        )
     else:
         ax[0].axis("off")
         ax[4].axis("off")
         ax[5].axis("off")
 
 
-def scatter_plot(
-    fig: Figure,
-    ax: plt.Axes,
-    *,
+def single_plot(
+    fig: plt.Figure,
+    ax: plt.axes,
     lon: np.array,
     lat: np.array,
     data: np.array,
     cmap: str = "viridis",
     norm: str | None = None,
     title: str | None = None,
+    scatter: bool = False,
 ) -> None:
-    """Lat-lon scatter plot: can work with arbitrary grids.
+    """Plot a single lat-lon map.
+
+    Plotting can be made either using scatter plot or Datashader(bin) plots.
+    By default it uses Datashader since it is faster and more efficient.
 
     Parameters
     ----------
-    fig : Figure
+    fig : _type_
```

Comment:
Can the type be readded?

Reply:
good catch! updating this now. For consistency I am updating all plt.Figure types to just Figure, and there a few more type so I am fixing those to have the correct typings